### PR TITLE
Add reflectometry GUI runs presenter tests

### PR DESCRIPTION
**Pull request by gemmaguest**
This PR adds missing unit tests for the Runs tab on the ISIS Reflectometry GUI. This includes the following main changes:

- Add a test to ensure the table presenter is updated when doing a transfer of runs from the search box.
- Add unit tests for the live data monitor and a `MockAlgorithmRunner` to support them.
- Generalise some supporting functions in `ParseKeyValueString` to help with converting between strings and maps when comparing results.
- Add unit tests for the new `ParseKeyValueString` behaviour and also split up the large existing tests for these functions into smaller units.

**To test:**

- Code review
- Ensure the parsing still works in the ISIS Reflectometry interface. This is used for the Options column and the stitch parameters:
  - Enter instrument `INTER`, run number `13460`, angle `0.5` and Options `ProcessingInstructions=4` on the Runs tab.
  - On the Experiment tab, enter the output stitch params as `Params=-0.02`
  - Back on the Runs tab, click Process.
- Ensure the live data monitor button still works on the interface:
  - Click Start Monitor and check that it attempts to start the live data monitor (it may not succeed if the instrument isn't currently running but should give a sensible error).

Refs #26512 	

*This does not require release notes* because **it concerns testing only**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by giovannidisiena** [comment]
Tested as described, handles bad input and live data monitoring works. Do extreme large/small values for angle and stitch params want to be limited/handled? These inputs will still produce data but could be considered out of range?

**Comment by gemmaguest** [comment]
> Tested as described, handles bad input and live data monitoring works. Do extreme large/small values for angle and stitch params want to be limited/handled? These inputs will still produce data but could be considered out of range?

Good point - we should check this with instrument scientists. It's out of scope for this PR though so I've added that to issue #23049

File: qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h [hunk 1]
```diff
@@ -49,6 +52,7 @@ class RunsPresenterTest : public CxxTest::TestSuite {
         m_searcher(nullptr), m_pythonRunner(), m_runNotifier(nullptr),
         m_runsTable(m_instruments, m_thetaTolerance, ReductionJobs()),
         m_searchString("test search string"), m_searchResult("", "", "") {
+    Mantid::API::FrameworkManager::Instance();
```
Comment (giovannidisiena) [inline qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h]: Just trying to understand what this does?

File: qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h [hunk 2]
```diff
@@ -646,12 +712,41 @@ class RunsPresenterTest : public CxxTest::TestSuite {
...
+    m_searchResult = SearchResult("", "", "");
     for (auto rowIndex : selectedRows)
       EXPECT_CALL(*m_searcher, getSearchResult(rowIndex))
           .Times(1)
           .WillOnce(ReturnRef(m_searchResult));
   }
 
+  // Set up a valid search result with content and return the corresponding
+  // model
+  ReductionJobs
```
Comment (giovannidisiena) [inline qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h]: What is this line doing?

**Review by giovannidisiena — APPROVED** [review]
Don't fully understand all the code here, especially mocking, however it looks good to me. Nice coverage on the tests and good commenting.

File: qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h (hunk 1 above)
Comment (gemmaguest) [reply qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h]: It initialises the framework, which is needed to be able to create algorithms in the tests.

File: qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h (hunk 2 above)
Comment (gemmaguest) [reply qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h]: It's just the return value - clang-format just saw fit to put it on a separate line!

Comment (giovannidisiena) [reply qt/scientific_interfaces/test/ISISReflectometry/Runs/RunsPresenterTest.h]: Oh yes, I see that now haha!

**Pull request merged**